tools/nodejs_api/src_cpp/each_single_async_worker.cpp

```diff
@@ -0,0 +1,35 @@
+#include "include/each_single_async_worker.h"
+
+#include <chrono>
+#include <thread>
+#include <vector>
+
+#include "include/node_query_result.h"
+#include "include/util.h"
+
+using namespace std;
+
+EachSingleAsyncWorker::EachSingleAsyncWorker(Function& callback, shared_ptr<kuzu::main::QueryResult>& queryResult, size_t index)
+    : AsyncWorker(callback), queryResult(queryResult), index(index) {};
+
+void EachSingleAsyncWorker::Execute() {
+    try {
+        auto row = queryResult->getNext();
+        for (size_t j = 0; j < row->len(); j++) {
+            rowResult.emplace_back(
+                make_unique<kuzu::common::Value>(kuzu::common::Value(*row->getValue(j))));
+        }
+    } catch(const std::exception &exc) {
+        SetError("Unsuccessful async Each callback: " + std::string(exc.what()));
+    }
+}
+
+void EachSingleAsyncWorker::OnOK() {
+    Napi::Array rowArray = Napi::Array::New(Env(), rowResult.size() + 1);
```

Comment:
I think it is actually possible to move this work from the main thread with `ThreadSafeFunction` (https://github.com/nodejs/node-addon-api/blob/main/doc/threadsafe_function.md). Please consider using it instead of performing the conversions in `OnOK`.